### PR TITLE
feat: rolemapping for gitlab groups

### LINKED ISSUE #39230: Map Gitlab Groups to Grafana Groups
Gitlab Users currently can only mapped based on their User Properties. These are not that broad to differentiate between a group of users. This would allow to map Grafana Roles based on Gitlab Groups. We in detail would need this as we provision a Grafana Instance per Kubernetes Cluster and want to map a group of gitlab users per k8s cluster.

### PR DESCRIPTION
**Pull request by henrysachs**
**What this PR does / why we need it**:

As discussed with @splattael in #28892 and #30025 we thought it would be nice not to only match against the user object, but also do queries against the groups we are requesting anyway for the `allowed groups` Feature. In my personal case the User Object itself doesn't deliver as much information as we would need to determine if a user should be Admin in Grafana. 

This MR allows you to write a JMES Array query to query all fields a Gitlab Group Offers. So you could for example say all users that are in Group 10485 should be Editor while all Gitlab Admins should be Admins could look like the following:

```bash
role_attribute_path = is_admin && 'Admin' || [?id==`10486`] && 'Editor'
```

As Groups are many and the current User is a single object the queries differentiate each other through how you would write them. I also implemented it so that if the query matches 

**Which issue(s) this PR fixes**:

Fixes #39230

**Special notes for your reviewer**:

This currently is my rough implementation and i want to clean up some things I just wanted to create the PR to discuss the Feature and also get some early Feedback

### CODE REVIEW

**Comment by henrysachs** [comment]
@marefr you had a look on the last PR and it seemed like you wanted to discuss this further. Here is the follow up pr i talked about in the other PR. :)

**Comment by henrysachs** [comment]
cleaned up some initial implementations I've done. Would be neat to discuss it now. :)

**Comment by sdboyer** [comment]
Sorry about that, @henrysachs.

Anybody have context here, @grafana/backend-platform ? @marefr ? I've got none 😔

**Review by marefr — COMMENTED** [review]
Added a quick comment. Lots of other things going on at the moment unfortunately

File: pkg/login/social/gitlab_oauth.go [hunk 1]
```diff
@@ -115,21 +131,40 @@ func (s *SocialGitlab) UserInfo(client *http.Client, token *oauth2.Token) (*Basi
...
+		groupFullNames[i] = group.FullPath
+	}
+
 	role, err := s.extractRole(response.Body)
 	if err != nil {
 		s.log.Error("Failed to extract role", "error", err)
 	}
 
+	// only evaluate groupqueries if userqueries haven't matched anything
+	if role == "Viewer" || role == "" {
```
[the remark this answers is not in the document]
Comment (henrysachs) [reply pkg/login/social/gitlab_oauth.go:145]: So I'm not sure if I can transition the Issue to a discussion but we already have to mentions from other people having interest in the feature itself.

**Pull request closed without merging**